src/api/app/components/confirmation_dialog_component.rb

```diff
@@ -0,0 +1,20 @@
+# This component accepts these parameters:
+# - modal_id: it is compulsory and should be unique, you have to pass it to the component and you can not use the
+#             same id in two different calls to the component.
+# - action: (path to the destroy action) you either pass it to the component or is '#' by default because we are going to set it by JavaScript.
+# - modal_title and confirmation_text: they are optional. Pass them only if you want to overwrite the default texts.
+# - method: pass it if you need a method different from :delete.
+
+class ConfirmationDialogComponent < ApplicationComponent
+  attr_accessor :modal_id, :action, :modal_title, :confirmation_text, :method
+
+  def initialize(modal_id:, options: {})
+    super
+
+    @modal_id = modal_id
+    @action = options[:action] || '#'
+    @modal_title = options[:modal_title] || 'Do you really want to remove this item?'
+    @confirmation_text = options[:confirmation_text] || 'Please confirm you want to remove this item.'
+    @method = options[:method] || :delete
```

Comment:
@saraycp I would be careful with defaulting to a `delete` action, once forgotten to set we might have a problem. IMO its not worth defaulting to anything here.

Reply:
Alright, I'm going to make the `method` parameter compulsory, so we are forced to consciously decide which method we want to use when we use the component.

Reply:
done